src/sage/crypto/public_key/key_exchange/commutative_key_exchange.py

```diff
@@ -0,0 +1,35 @@
+from sage.crypto.public_key.key_exchange import KeyExchange
+from abc import ABC, abstractmethod
+
+class CommutativeKeyExchange(KeyExchange, ABC):
+
+    @abstractmethod
+    def secret_key(self):
+        raise NotImplementedError
+
+    @abstractmethod
+    def public_key(self, secret_key):
+        raise NotImplementedError
+
+    @abstractmethod
+    def compute_shared_secret(self, sk, pk):
+        raise NotImplementedError
+    
+    def alice_secret_key(self):
+        return self.secret_key()
+
+    def alice_public_key(self, alice_secret_key):
+        return self.public_key(alice_secret_key)
+
+    def bob_secret_key(self):
+        return secret_key()
+
+    def bob_public_key(self, bob_secret_key):
+        return self.public_key(bob_secret_key)
+
+    def alice_compute_shared_secret(self, alice_sk, bob_pk):
+        return self.compute_shared_secret(alice_sk, bob_pk)
+
+    def bob_compute_shared_secret(self, bob_sk, alice_pk):
+        return self.compute_shared_secret(bob_sk, alice_pk)
+
```

Comment:
It would be nice to to exclude the `alice_foo` and `bob_foo` versions from tab completion for easier discoverability. You can do this with the dunder method `__dir__`. Make sure you write it in a way that will not exclude new methods implemented in subclasses though (i.e. just exclude the 6 methods that are implemented here).